cmscontrib/loaders/lio.py

```diff
@@ -0,0 +1,322 @@
+                )
+                args['statements'][lang] = Statement(lang, digest)
+            if args['statements']:
+                args['primary_statements'] = self.conf.get('primary_statements', ["lv"])
+
+        args['submission_format'] = [f"{name}.%l"]
+
+        set_if_present(self.conf, args, 'score_precision')
+
+        score_mode = self.conf.get('score_mode', SCORE_MODE_MAX_TOKENED_LAST)
+        if score_mode in [SCORE_MODE_MAX, SCORE_MODE_MAX_SUBTASK, SCORE_MODE_MAX_TOKENED_LAST]:
+            args['score_mode'] = score_mode
+        else:
+            logger.critical("Unknown score mode provided")
+            raise ValueError
+
+        set_if_present(self.conf, args, 'max_submission_number', default=40)
+        set_if_present(self.conf, args, 'max_user_test_number', default=40)
+        set_if_present(self.conf, args, 'min_submission_interval', make_timedelta)
+        set_if_present(self.conf, args, 'min_user_test_interval', make_timedelta)
+
+        task = Task(**args)
+
+        args = {}
+        args["task"] = task
+        args["description"] = self.conf.get("version", "Default")
+        args["autojudge"] = False
+
+        args['time_limit'] = float(self.conf.get('time_limit', 2))
+        # Memory limit in MiB
+        args['memory_limit'] = self.conf.get('memory_limit', 256) * 1024**2
+
+        # Builds the parameters that depend on the task type
+        args["managers"] = {}
+
+        # By default use standard input, output
+        input_filename = self.conf.get("input_filename", "")
+        output_filename = self.conf.get("output_filename", "")
+
+        # No grader support
+        compilation_param = "alone"
+
+        if 'checker' in self.conf:
+            logger.info("Checker found, compiling")
+            checker_src = os.path.join(self.task_dir, self.conf['checker'])
+            checker_exe = os.path.join(os.path.dirname(checker_src), "checker")
+            if config.installed:
+                testlib_path = "/usr/local/include/cms"
+            else:
+                testlib_path = os.path.join(os.path.dirname(__file__), "polygon")
+            code = subprocess.call(["g++", "-x", "c++", "-O2", "-static",
+                                    "-DCMS", "-I", testlib_path,
+                                    "-o", checker_exe, checker_src])
+            if code != 0:
+                logger.critical("Could not compile checker")
+                return None
+            digest = self.file_cacher.put_file_from_path(
+                checker_exe, "Checker for task {name}"
+            )
+            args["managers"]["checker"] = Manager("checker", digest)
+            evaluation_param = "comparator"
+        else:
+            evaluation_param = "diff"
+
+        point_file = os.path.join(self.task_dir, self.conf.get('point_file', 'punkti.txt'))
+        points_per_group = self.parse_point_file(point_file)
+        if sum(points_per_group.values()) != 100:
```

Comment:
Should this be moved into `parse_point_file`?

Reply:
Done